plutus-core/executables/debugger/Event.hs

```diff
@@ -1,20 +1,44 @@
-{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
 
 -- | Handler of debugger events.
 module Event where
 
+import Annotation
 import Types
+import UntypedPlutusCore.Evaluation.Machine.Cek.Debug.Driver qualified as D
 
 import Brick.Focus qualified as B
 import Brick.Main qualified as B
 import Brick.Types qualified as B
 import Brick.Widgets.Edit qualified as BE
+import Control.Concurrent.MVar
 import Control.Monad.State
+import Data.MonoTraversable
 import Graphics.Vty qualified as Vty
 import Lens.Micro
 
-handleDebuggerEvent :: B.BrickEvent ResourceName e -> B.EventM ResourceName DebuggerState ()
-handleDebuggerEvent bev@(B.VtyEvent ev) = do
+type Breakpoints = [Breakpoint]
+data Breakpoint = UplcBP SourcePos
+                | TxBP SourcePos
```

Comment:
`TxBP` -> `SourceBP` (or something else)
The source is not necessarily PlutusTx. Other source languages can also be used with the debugger.